apollo-federation/src/query_graph/build_query_graph.rs

```diff
@@ -2127,6 +2131,7 @@ impl FederatedQueryGraphBuilder {
                     schema,
                     type_in_supergraph_pos.type_name.clone(),
                     "__typename",
+                    false,
```

Comment:
Why do we not validate here?

Reply:
In this case, the field set (`__typename`) is trivially valid on `type_in_supergraph_pos` (which we guarantee is an interface in the preceding statement). I can make that clearer by adding a comment there, so at least if it gets changed in the future they'll remember to flip the boolean to `true`.